src/client/src/commons/map/pointComponent.js

```diff
@@ -210,58 +200,71 @@ class PointComponent extends React.Component {
     }
   }
 
+  drawOrUpdatePoint(point) {
+    if (this.centerFeature) {
+      this.centerFeature.getGeometry().setCoordinates(point);
+    } else {
+      this.centerFeature = new Feature({
+        name: "Center",
+        geometry: new Point(point),
+      });
+      this.position.addFeature(this.centerFeature);
+      this.map.getView().setResolution(1);
+      this.map.getView().setCenter(point);
+    }
+  }
+
+  removeMapFeature() {
+    this.centerFeature = null;
+    this.position.clear();
+    this.setState({
+      ...this.state,
+      point: null,
+    });
+  }
+
   manageMapInteractions() {
     const { x, y, isLocked } = this.props;
+    const point = x && y ? [x, y] : null;
     if (isLocked) {
-      // add ol modify point interaction.
-      this.modify = new Modify({
-        source: this.position,
-      });
-      // add ol draw point interaction.
-      this.draw = new Draw({
-        type: "Point",
-        source: this.position,
-      });
-      this.map.addInteraction(this.draw);
-      this.map.addInteraction(this.modify);
-
-      // deactivate draw if a point already exists.
-      const point = x && y ? [x, y] : this.state.point ?? null;
       if (point) {
-        this.centerFeature = new Feature({
-          name: "Center",
-          geometry: new Point(point),
-        });
-        this.map.getView().setResolution(1);
-        this.map.getView().setCenter(point);
-        this.position.addFeature(this.centerFeature);
-        this.draw && this.draw.setActive(false);
+        this.drawOrUpdatePoint(point);
+        this.allowModifying();
```

Comment:
Q: Hier ist `isLocked = true` jedoch wird `allowModifying()` aufgerufen. Ist das korrekt?

Reply:
Ja , das ist ein Problem mit den etwas veralteten Begrifflichkeiten in der App.

Wenn das Borehole editierbar ist, ist es "locked" im Sinne von "durch dich für andere Benutzer gesperrt". 
Ist es nicht editierbar ist es "not locked", wie in der Fehlermeldung unten. 

Vielleicht ist es aber sinnvoll im Code weniger verwirrende Begriffe zu verwenden.

![image](https://user-images.githubusercontent.com/49144408/210731034-b3cd1b80-8078-4d37-a049-6ece6548fd2b.png)